### ISSUE TITLE
A problem of the scaling of the Q, K inner product

### ISSUE THREAD

**Issue by b06b01073** [FrancescoSaverioZuppichini/ViT#6]
First of all, thank you for this awesome source code!!!

My question is regarding the `forward` function call in the `MultiHeadAttention` class, there's an `F.softmax` call which I believe is inconsistent with the original Transformer paper, where the inner product of Q and K is scaled first ([paper](https://arxiv.org/pdf/1706.03762.pdf), equation (1)) then comes the softmax. In your code, `att = F.softmax(energy, dim=-1) * self.scaling` was written, and I'm wondering whether this is a typo or there's some specific reason for this? Thank you.
